projects/igniteui-angular/src/lib/progressbar/progressbar.component.ts

```diff
@@ -476,7 +477,10 @@ export class IgxLinearProgressBarComponent extends BaseProgress {
 
 @Component({
     selector: 'igx-circular-bar',
-    templateUrl: 'templates/circular-bar.component.html'
+    templateUrl: 'templates/circular-bar.component.html',
+    providers: [
+        Directionality
+    ]
```

Comment:
Leave the service in the root injector provider. This will create a separate service for each progress bar component.